### PR TITLE
Add sse4/avx2 support for fast x86 int8 (vpmaddubsw/vpmaddwd/vpaddd)

### PR DESCRIPTION
**Pull request by elvin-n**
- Extend the list of different target for x86 topi
- Extend tests for conv2d x86 int8 for fast i8 x86 platforms

this change in theory can give up to 2x speedup on int8 models vs fp32 models, currently slightly less

Resnet50 performance:
<html><body>
<google-sheets-html-origin><style type="text/css"></style>

  | Core i7-1185G7 sse4 | Core i7-1185G7 avx2 | Core i7-1185G7 avx512 | Core i7-1185G7 VNNI | Core i7-8700B | Core i5-9400T
-- | -- | -- | -- | -- | -- | --
  | FPS | FPS | FPS | FPS | FPS | FPS
TVM FP32 |   | 53 | 53 | 53 | 54 | 48
TVM int32 |   | 12 |   |   | 16 |  
TVM int8 default | 34 | 61 | 92 | 142 | 78 | 62
TVM int8 atvm |   | 70 |   | 134 | 95 | 79

</body>
</html>

### CODE REVIEW

**Review by vinx13 — APPROVED** [review]

File: python/tvm/topi/x86/utils.py [hunk 1]
```diff
@@ -18,9 +18,95 @@
 import tvm
 
 
-def get_fp32_len():
+def target_has_sse42(target):
```
Comment (jcf94) [inline python/tvm/topi/x86/utils.py:21]: Just curious about why it's named `sse42` ... Is it for `sse4 & avx2` like the pr title, or minor version like `sse 4.2`?

And an unimportant suggestion which you can ignore is to merge all of these functions below to something like `target_has_attr(target, attr)` and list the candidates of attr in the doc string.

File: tests/python/relay/test_op_level2.py [hunk 2]
```diff
@@ -1687,7 +1692,7 @@ def _has_fast_int8_instructions(asm, target):
             dtypes=fast_int8_dtypes,
         )
         # Check that vector int mult and add instructions are generated.
-        assert "vpmulld" in asm and "vpadd" in asm
+        assert "pmulhw" in asm and "paddd" in asm
```
Comment (jcf94) [inline tests/python/relay/test_op_level2.py:1695]: I'm not so familiar the specific instructions, does `pmulhw` and `paddd` still be vectorize instructions in this test?

File: python/tvm/topi/x86/utils.py (hunk 1 above)
Comment (elvin-n) [reply python/tvm/topi/x86/utils.py:21]: sse4.2 is a latest standard of SSE type of instructions and supported in more processors than sse4/sse3 or sse2. It continues to be used in the latest Intel edge devices for IoT or low end segments (Atom based). If we come to the requirement distinguish more precisely, we probably will have to redesign this part.

as for suggestion to introduce the only function instead several ones - I consider this as valuable comment and in the the future if we want to check more features this should done. Not sure that it make sense to do in this PR

File: tests/python/relay/test_op_level2.py (hunk 2 above)
Comment (elvin-n) [reply tests/python/relay/test_op_level2.py:1695]: It is vector instruction from the middle of 90'th :) It is MMX working on 64 bits

**Review by jcf94 — APPROVED** [review]
Thanks for your answer! @elvin-n

**Comment by elvin-n** [comment]
The change in get_fp32_len affected ARM flow - now it started to block by 4 instead previous default 8. It must not affect from performance point of view since NEON SIMD vector size if 64 or 128 bit, but will affect the knowledge database of tuned kernels.

Will verify the performance aspect on ARM. As for backward compatibility - still open question. So far I have an impression that we do not care about it so much.

**Comment by elvin-n** [comment]
I verified ARM flow and confirm that it started to use 4 channel values instead of 8 for blocking and this fact did not affect performance  anyhow (as i expected)

**Pull request merged**